lock-api/src/main/java/com/palantir/lock/watch/LockWatchCacheImpl.java

```diff
@@ -43,14 +43,20 @@ public void processStartTransactionsUpdate(Set<Long> startTimestamps, LockWatchS
     public void processCommitTimestampsUpdate(
             Collection<TransactionUpdate> transactionUpdates, LockWatchStateUpdate update) {
         eventCache.processGetCommitTimestampsUpdate(transactionUpdates, update);
-        valueCache.updateCacheOnCommit(
+        valueCache.updateCacheWithCommitTimestampsInformation(
                 transactionUpdates.stream().map(TransactionUpdate::startTs).collect(Collectors.toSet()));
     }
 
     @Override
-    public void removeTransactionStateFromCache(long startTimestamp) {
-        eventCache.removeTransactionStateFromCache(startTimestamp);
-        valueCache.removeTransactionState(startTimestamp);
+    public void removeTransactionStateFromCache(long startTs) {
+        eventCache.removeTransactionStateFromCache(startTs);
+        valueCache.ensureStateRemoved(startTs);
+    }
+
+    @Override
+    public void onSuccess(long startTs) {
+        valueCache.onSuccessfulCommit(startTs);
+        eventCache.removeTransactionStateFromCache(startTs);
```

Comment:
Note the reversed order here - it matters for `onSuccess`, but actually doesn't matter for `removeTransactionStateFromCache`